### PR TITLE
Add CI Support

### PR DESCRIPTION
**Pull request by clearydude**
Prior to this pull request, the OSCC-based joystick commander application wasn't building on any CI pipeline. This commit adds a simple Jenkinsfile to automatically trigger new jc builds when pushes occur.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  try {
+	stage('Checkout') {
+	  clean_checkout()
+	}
+    stage('Build') {
+        sh 'git clone git@github.com:PolySync/oscc.git --branch master && mkdir build && cd build && cmake .. -DKIA_SOUL=ON && make'
+        echo 'Build Complete!'
+    }
+  }
+  catch(Exception e) {
```
Comment (adrienthebo) [inline Jenkinsfile]: Does the code in the `try` block potentially throw a checked exception? If not, can we elide this `catch` block?

Comment (clearydude) [reply Jenkinsfile]: Any of the commands in the shell script could *potentially* fail, so I want to make sure the directory isn't deleted. I'm not sure if the `catch` block is specifically needed, but I think the finally block should certainly be there to ensure we cleanup.

Comment (rebpdx) [reply Jenkinsfile]: I was a skeptic about dropping the catch but found it to work from a similar comment when ROSCCO was being reviewed. https://github.com/PolySync/roscco/blob/devel/Jenkinsfile Apparently try/finally is a thing...

Comment (clearydude) [reply Jenkinsfile]: Alrighty! I'll try it without.

**Review by adrienthebo — APPROVED** [review]
I have only the style nitpick about `try/catch` blocks being used to re-raise an exception but that's of little consequence. If we get this change in then fantastic but I don't want something small like that to block this PR. :+1:

File: Jenkinsfile (hunk 1 above)
Comment (clearydude) [reply Jenkinsfile]: Removed as of 2f1377c

**Review by rebpdx — APPROVED** [review]

**Pull request merged**